networkx/algorithms/traversal/depth_first_search.py

```diff
@@ -192,26 +245,33 @@ def dfs_postorder_nodes(G,source=None):
     Examples
     --------
     >>> G = nx.Graph()
-    >>> G.add_path([0,1,2])
-    >>> print(list(nx.dfs_postorder_nodes(G,0)))
-    [2, 1, 0]
+    >>> G.add_path([0,1,2,3,4])
+    >>> list(nx.dfs_postorder_nodes(G,0))
+    [4, 3, 2, 1, 0]
+
+    >>> list(nx.dfs_postorder_nodes(G,0,2))
+    [1, 0]
 
     Notes
     -----
-    Based on http://www.ics.uci.edu/~eppstein/PADS/DFS.py
-    by D. Eppstein, July 2004.
-
     If a source is not specified then a source is chosen arbitrarily and
     repeatedly until all components in the graph are searched.
+
+    The implementation of this function is adapted from David Eppstein's
+    depth-first search function in `PADS`_, with modifications
+    to allow depth limits based on the Wikipedia article
+    "`Depth-limited search`_".
+
+    .. _PADS: http://www.ics.uci.edu/~eppstein/PADS
+    .. _Depth-limited search: https://en.wikipedia.org/wiki/Depth-limited_search
     """
-    post=(v for u,v,d in nx.dfs_labeled_edges(G,source=source)
-          if d['dir']=='reverse')
+    edges = nx.dfs_labeled_edges(G, source=source, depth_limit=depth_limit)
     # potential modification: chain source to end of post-ordering
     # return chain(post,[source])
```

Comment:
You've removed "post" from the code so you either need to update this comment or remove it.